File: src/daedalus/network/node/LookupRequest.java
package daedalus.network.node;

import daedalus.network.protocol.InterestPacket;

class LookupRequest extends Request {
	private InterestPacket packet; 
	
	public LookupRequest(NetworkNode sender, InterestPacket packet) {
		super(sender);
		this.packet = packet;
	}

	public InterestPacket getPacket() {
		return packet;
	}

	public void setPacket(InterestPacket packet) {
		this.packet = packet;
	}
	
	@Override
	public String getPacketId() {
		return this.packet.getDataName();
	}
}

File: src/network/node/lookup_request.cpp
#include "./lookup_request.hpp"

network::node::lookup_request::lookup_request(network_node* sender, network::protocol::interest_packet packet)
    : request(sender) {

    this->m_packet = packet;
}

network::protocol::packet const& network::node::lookup_request::packet() {
    return this->m_packet;
}